apps/confidential/confidential.go

```diff
@@ -399,11 +399,11 @@ func New(clientID string, cred Credential, options ...Option) (Client, error) {
 		// Declaring the authority host known prevents unnecessary metadata discovery requests. (The
 		// authority is irrelevant to Client and friends because the token provider is responsible
 		// for authentication.)
-		parsed, err := url.Parse(opts.Authority)
+		_, err := url.Parse(opts.Authority)
```

Comment:
This change makes parsing the authority and most of the comment vestigial. This block can simply be
```go
// The caller will handle all details of authentication, using Client only as a token cache.
baseOpts = append(baseOpts, base.WithInstanceDiscovery(false))
```